railties/lib/rails/generators.rb

```diff
@@ -156,12 +156,25 @@ def self.invoke(namespace, args=ARGV, config={})
         args << "--help" if args.empty? && klass.arguments.any?(&:required?)
         klass.start(args, config)
       else
-        options     = sorted_groups.map(&:last).flatten
-        suggestions = options.sort_by {|suggested| levenshtein_distance(namespace.to_s, suggested) }.first(3)
-        msg =  "Could not find generator '#{namespace}'. "
-        msg << "Maybe you meant #{ suggestions.map {|s| "'#{s}'"}.join(" or ") }\n"
-        msg << "Run `rails generate --help` for more options."
-        puts msg
+        unless sorted_groups.group_by(&:size).min.first > namespace.length
+          options     = sorted_groups.map(&:last).flatten
+          suggestions = options.select do |option|
+            option.split(/[:_]/).any? { |option_segment|
+              namespace.to_s.split(/[:_]/).any? { |name_segment|
```

Comment:
We're splitting over `:` here and then matching against each segment.

That means if someone is searching for `coffee:assetss` it will split it up to `["coffee", "assetss"] and then each of those will be checked for Levenshtein distance against all split generators. So "coffee" would have a levenshtein distance of 0 to "coffee", that means it will return all suggestions with `"coffee"` prefix and since you're only selecting the first one, if there's more than one generator with that prefix it will probably be wrong.